hotshot-task-impls/src/helpers.rs

```diff
@@ -489,17 +492,32 @@ pub async fn decide_from_proposal<TYPES: NodeType>(
         tracing::debug!("Leaf ascension failed; error={e}");
     }
 
-    if with_epochs && res.new_decided_view_number.is_some() {
-        let epoch_height = consensus_reader.epoch_height;
-        drop(consensus_reader);
+    let epoch_height = consensus_reader.epoch_height;
+    drop(consensus_reader);
 
+    if with_epochs && res.new_decided_view_number.is_some() {
         if let Some(decided_leaf_info) = res.leaf_views.last() {
             decide_epoch_root(&decided_leaf_info.leaf, epoch_height, membership).await;
         } else {
             tracing::info!("No decided leaf while a view has been decided.");
         }
     }
 
+    if let (Some(decided_upgrade_cert), Some(decided_leaf_info)) =
+        (&res.decided_upgrade_cert, res.leaf_views.last())
+    {
+        if decided_upgrade_cert.data.new_version == V::Epochs::VERSION {
+            let decided_block_number = decided_leaf_info.leaf.block_header().block_number();
+            let first_epoch_number =
+                TYPES::Epoch::new(epoch_from_block_number(decided_block_number, epoch_height));
```

Comment:
I thought the plan was to have the block height configured for the first epoch, but his code is getting the epoch from when we decide the upgrade.  Is that change for a different PR or am I misunderstanding/misremembering?